schemaRegistryClient.go

```diff
@@ -693,15 +693,22 @@ func cacheKey(subject string, version string) string {
 	return fmt.Sprintf("%s-%s", subject, version)
 }
 
+type Error struct {
+	Code    int    `json:"error_code"`
+	Message string `json:"message"`
+	str     *bytes.Buffer
+}
+
+func (e Error) Error() string {
+	return e.str.String()
+}
+
 func createError(resp *http.Response) error {
-	decoder := json.NewDecoder(resp.Body)
-	var errorResp struct {
-		ErrorCode int    `json:"error_code"`
-		Message   string `json:"message"`
-	}
-	err := decoder.Decode(&errorResp)
-	if err == nil {
-		return fmt.Errorf("%s: %s", resp.Status, errorResp.Message)
+	err := Error{str: bytes.NewBuffer(make([]byte, 0, resp.ContentLength))}
+	decoder := json.NewDecoder(io.TeeReader(resp.Body, err.str))
+	marshalErr := decoder.Decode(&err)
+	if marshalErr == nil {
+		return err
```

Comment:
Could you switch return err and return fmt.Errorf("%s", resp.Status)? I just think main case should be at the end of the method rather than inside an if statement.

Reply:
fixed